fakestagram/src/shared/hooks/useApi.js

```diff
@@ -0,0 +1,29 @@
+import { useCallback } from "react";
+import axios from "axios";
+
+const useApi = () => {
+  const doRequest = useCallback(async (url, method = 'GET', data = null, requiresToken = false) => {
+    let headers = {
+      "Content-Type": "application/json",
+    };
+    if (requiresToken) {
+      const user = JSON.parse(localStorage.getItem('user'));
+      const token = user?.token;
+      if (token) {
+        headers['Authorization'] = `Bearer ${token}`;
+      }
+    }
+
+    const response = await axios({
+      url: `http://64.23.228.143:3001/api/${url}`,
+      method,
+      headers,
+      data,
+    });
+    return response;
+  }, []);
+
+  return { doRequest };
+};
```

Comment:
we could take into consideration when we send images, as the Content-Type changes, because in the future we are going to send images (I guess). We can do something like:

    if (config.data instanceof FormData) {
      delete config.headers["Content-Type"];
    }
    
    This is not optimal, but it will work